LoopKitUI/Views/FractionalQuantityPicker.swift

```diff
@@ -148,6 +151,81 @@ struct FractionalQuantityPicker: View {
         }
     }
 
+    private var wholeGuardrail: Guardrail<HKQuantity> {
+        // Whether or not the whole value should be colored as 'out of range' depends on the fraction value
+        let lowerRecommendedBound = guardrail.recommendedBounds.lowerBound.doubleValue(for: unit)
+        if fraction < lowerRecommendedBound.fraction {
```

Comment:
[nit] is this assuming the lower recommended bound always only has a fractional component (e.g., 0.xxxx)?

Reply:
It's not, no. The coloration of the whole value is dependent only upon the fractional part, though; e.g. if carb ratio warnings start at 3 g/U, the '3' should be highlighted yellow if '00' is selected on the fractional side, regardless of what the current selected whole value is.